File: WebApi/Entities/Artist.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using WebApi.Services;

namespace WebApi.Entities
{
    public class Artist : IdentifiableEntity
    {
        [Filter(AllowedFilters = FilterType.EqualsTo | FilterType.Like)]
        public string Name { get; set; }

        public List<Album> Albums { get; set; }
    }
}

File: Client/src/app/dtos/artist.ts
import { AlbumDtoShort } from "./album";

export class ArtistDtoShort { 
    id!: number;
    name!: string;
}

export class ArtistDtoExtended { 
    id!: number;
    name!: string;

    albums!: AlbumDtoShort[];
}